omop_file_validator.py

```diff
@@ -438,13 +522,171 @@ def run_checks(file_path, f):
+                               nrows=1,
+                               lines=True,
+                               convert_dates=False)
+
+            if is_line_blank(row.iloc[0]):
+                blank_lines_msg = f'File contains blank line on line {idx}.'
+
+                result['errors'].append(dict(message=blank_lines_msg))
+                row_error_found = True
+
+            # check columns if looks good process file
+            if not _check_columns(cdm_column_names, row.columns, result):
+                row_error_found = True
+
+            #search for scientific notation
+            int_columns = [
+                col['name'] for col in cdm_table_columns
+                if col['type'] == 'integer'
+            ]
+
+            sci_not_line = has_scientific_notation_error(row, int_columns)
+            for col, (value, line_num) in sci_not_line.items():
+                sci_not_error_msg = dict(message=(
+                    f"Scientific notation value '{value}' was found on line {idx}. "
+                    "Scientific notation is not allowed for integer fields."),
+                                         column_name=col)
+                result['errors'].append(sci_not_error_msg)
+                row_error_found = True
+
+            for meta_item in cdm_table_columns:
+                meta_column_name = meta_item['name']
+                meta_column_required = meta_item['mode'] == 'required'
+                meta_column_type = meta_item['type']
+
+                if meta_column_name not in row.columns and meta_column_required:
+                    result['errors'].append(
+                        dict(message='Missing required column',
+                             column_name=meta_column_name))
+                    row_error_found = True
+                elif pd.isnull(
+                        row[meta_column_name].loc[0]) and meta_column_required:
+                    result['errors'].append(
+                        dict(message=MSG_NULL_DISALLOWED,
+                             column_name=meta_column_name))
+                    row_error_found = True
+                else:
+                    value = row[meta_column_name].loc[0]
+                    row_column_type = row[meta_column_name].dtype
+
+                    if find_error_in_row(row, meta_column_name,
+                                         meta_column_type):
+                        e = dict(message=MSG_INVALID_TYPE + " line number " +
+                                 str(idx + 1),
+                                 column_name=meta_column_name,
+                                 actual=value,
+                                 expected=meta_column_type)
+                        result['errors'].append(e)
+                        row_error_found = True
+
+                    # Check that date format is in the YYYY-MM-DD or YYYY-MM-DD hh:mm:ss format
+                    if meta_column_type in ('date', 'timestamp'):
+                        fmts = ''
```

Comment:
I think this should be the same in run_csv_checks as well. It is currently `fmt = ''`